### PR TITLE
eth: update higest block we know during the sync if a higher was found

### PR DESCRIPTION
**Pull request by rjl493456442**
In this PR, fix two problems:

1. Avoid useless synchronization during the fast sync if the peer doesn't have higher difficulty than our fast block head.

2. Update highest block we know during the sync

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func (s *PublicEthereumAPI) Syncing() (interface{}, error) {
 	progress := s.b.Downloader().Progress()
 
 	// Return not syncing if the synchronisation already completed
-	if progress.CurrentBlock >= progress.HighestBlock {
+	if !s.b.Downloader().Synchronising() {
```
Comment (karalabe) [inline internal/ethapi/api.go]: This won't be a good change:

```
// Synchronising returns whether the downloader is currently retrieving blocks.
func (d *Downloader) Synchronising() bool {
	return atomic.LoadInt32(&d.synchronising) > 0
}
```

During the node's initial sync (and maybe even afterwards), sync will abort and restart a few times if the master peer we're connected to disappears. The current code `progress.CurrentBlock >= progress.HighestBlock` doesn't care about that, and will just report syncing until we're actually in sync. Your change would report sync done whenever the downloader aborts for a few second.

**Comment by rjl493456442** [comment]
@karalabe Updated.

**Comment by rjl493456442** [comment]
@karalabe Please take a look again

File: eth/sync.go [hunk 2]
```diff
@@ -185,9 +185,15 @@ func (pm *ProtocolManager) synchronise(peer *peer) {
 		// The only scenario where this can happen is if the user manually (or via a
 		// bad block) rolled back a fast sync node below the sync point. In this case
 		// however it's safe to reenable fast sync.
+
+		// Make sure the peer's total difficulty we are synchronizing is higher.
+		if pm.blockchain.GetTdByHash(pm.blockchain.CurrentFastBlock().Hash()).Cmp(pTd) >= 0 {
+			return
+		}
```
Comment (karalabe) [inline eth/sync.go]: This is only one path where fast sync is enabled, the above if branch also enables fast sync.

Please move these two lines below outside of both branches, and do something like

```go
if mode == downloader.FastSync {
	if pm.blockchain.GetTdByHash(pm.blockchain.CurrentFastBlock().Hash()).Cmp(pTd) >= 0 {
		return
	}
}
```

**Pull request merged**